src/main/webapp/app/tasks/services/local-task.service.ts

```diff
@@ -2,18 +2,28 @@ import { Injectable } from '@angular/core';
 import { Observable, of } from 'rxjs';
 import { v4 as uuid } from 'uuid';
 
-import { Task } from './task';
-import { TaskService } from './task.service';
+import { Task } from '../task';
+import { TaskService } from '../task.service';
+
+enum LogicalOperators {
+  AND = 'AND'
+}
 
 @Injectable()
 export class LocalTaskService implements TaskService {
-
   private static readonly STORAGE_KEY: string = 'tiny.tasks';
 
   getAll(): Observable<Task[]> {
     return of(this.readTasks());
   }
 
+  getByQuery(query: string): Observable<Task[]> {
+    const terms = query.split(LogicalOperators.AND).map(x => x.trim());
```

Comment:
this is quite clever because it also works with` r AND a AND b AND o` for example

Reply:
1. Yes, it works with multiple `AND` operators out-of-the-box :)

2. > I would hope for more elaborate variable naming than x
I agree with up to a point. Usually it makes sense to have a comprehensive clear variable name. 
But in this particular case, the map function remove empty spaces only, for every stuff, does not matter how is is called there.  

3. I did not implemented other operators as it would take more time (I mean combination of operators with different types)